packages/nextjs/pages/[contractAddress]/[network].tsx

```diff
@@ -1,18 +1,99 @@
+import { ParsedUrlQuery } from "querystring";
+import { Abi, isAddress } from "viem";
+import * as chains from "viem/chains";
+import { ExclamationTriangleIcon } from "@heroicons/react/24/outline";
+import { Footer } from "~~/components/Footer";
+import { MiniHeader } from "~~/components/MiniHeader";
+import { Spinner } from "~~/components/assets/Spinner";
+import { ContractUI } from "~~/components/scaffold-eth";
+import { useAbiNinjaState } from "~~/services/store/store";
+import { fetchContractABIFromEtherscan } from "~~/utils/abi";
+
+interface ParsedQueryContractDetailsPage extends ParsedUrlQuery {
+  contractAddress: string;
+  network: string;
+}
+
+type ContractData = {
+  abi: Abi;
+  address: string;
+};
 
 const ContractDetailPage = () => {
   const router = useRouter();
-  const { contractAddress, network } = router.query;
+  const { contractAddress, network } = router.query as ParsedQueryContractDetailsPage;
+  const [contractData, setContractData] = useState<ContractData>({ abi: [], address: contractAddress });
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const contractName = contractData.address;
+  const { contractAbi: storedAbi, setMainChainId } = useAbiNinjaState(state => ({
+    contractAbi: state.contractAbi,
+    setMainChainId: state.setMainChainId,
+  }));
+
+  const getNetworkName = (chainId: string) => {
+    const chain = Object.values(chains).find(chain => chain.id === parseInt(chainId));
+    return chain ? chain.name : "Unknown Network";
+  };
 
-  // You can now use contractAddress and network to fetch data, etc.
-  // For example, fetch contract details from an API or local data source
+  useEffect(() => {
+    if (network) {
+      setMainChainId(parseInt(network));
+    }
+  }, [network, setMainChainId]);
+
+  useEffect(() => {
+    const fetchContractAbi = async () => {
+      setIsLoading(true);
+      try {
+        if (storedAbi && storedAbi.length > 0) {
+          setContractData({ abi: storedAbi, address: contractAddress });
+        } else {
+          const abiString = await fetchContractABIFromEtherscan(contractAddress, parseInt(network));
+          const abi = JSON.parse(abiString);
+          setContractData({ abi, address: contractAddress });
+        }
+        setError(null);
+      } catch (e: any) {
+        setError(e.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    if (isAddress(contractAddress)) {
+      fetchContractAbi();
+    }
+  }, [contractAddress, network, storedAbi]);
 
   return (
-    <div>
-      <h1>Contract Details</h1>
-      <p>Contract Address: {contractAddress}</p>
-      <p>Network: {network}</p>
-      {/* Render additional contract details here */}
+    <div className="bg-base-100 min-h-screen">
+      <MiniHeader />
+      <div className="flex flex-col gap-y-6 lg:gap-y-8 justify-center items-center">
+        {isLoading ? (
+          <Spinner />
```

Comment:
@Pabl0cks should we use : 

```tsx
<span className="loading loading-spinner text-primary h-14 w-14"></span>
```

Or maybe `<Spinner height="100" width="100"  />` so that it looks a bit big ? 

Also we updated <Spinner /> component internal colors 

lol sorry for nitpick we can always improve this later

Reply:
I think not using Spinner is more consistent with `index.tsx` behavior, thanks for the feedback ♥

Agree it was also too small :) Just pushed your suggestion 🙌